File: RadarSDK/RadarConfig.m
//
//  RadarConfig.m
//  RadarSDK
//
//  Created by Nick Patrick on 1/4/23.
//  Copyright © 2023 Radar Labs, Inc. All rights reserved.
//

#import "RadarConfig.h"
#import "RadarMeta.h"
#import <Foundation/Foundation.h>

@implementation RadarConfig

+ (RadarConfig *)fromDictionary:(NSDictionary *)dict {
    RadarConfig *config = [RadarConfig new];

    if (dict) {
        id metaObj = dict[@"meta"];
        if (metaObj && [metaObj isKindOfClass:[NSDictionary class]]) {
            config.meta = [RadarMeta fromDictionary:metaObj];
        }

        id challengeObj = dict[@"nonce"];
        if (challengeObj && [challengeObj isKindOfClass:[NSString class]]) {
            config.challenge = (NSString *)challengeObj;
        }
    }

    return config;
}

@end

File: RadarSDK/RadarConfig.swift
//
//  RadarConfig.swift
//  RadarSDK
//
//  Copyright © 2026 Radar Labs, Inc. All rights reserved.
//

import Foundation

@objc(RadarConfig) @objcMembers
class RadarConfig: NSObject {
    var meta: RadarMeta?
    var nonce: String?

    static func from(dictionary dict: [String: Any]?) -> RadarConfig? {
        let config = RadarConfig()

        if let dict {
            if let metaDict = dict["meta"] as? [String: Any] {
                config.meta = RadarMeta.from(dictionary: metaDict)
            }
            if let nonce = dict["nonce"] as? String {
                config.nonce = nonce
            }
        }

        return config
    }
}
